corehq/util/couch_doc_processor.py

```diff
@@ -94,7 +94,6 @@ def __iter__(self):
 
         # save iteration state without offset to signal completion
         self.state.pop("offset", None)
-        self.state["retry"] = {}
```

Comment:
Might be easier to go to voice but I don't think this is true "If a retry
doc cannot be processed due to too many retries or some other problem that
prevents the iteration from completing, then it will be left in the retry
queue". Once the iteration is complete the retry queue is cleared (by the
line I removed) so any future runs will not retry those documents as
illustrated by this test case:
https://github.com/dimagi/commcare-hq/blob/master/corehq/util/tests/test_couch_helpers.py#L105